airflow/ti_deps/deps/not_previously_skipped_dep.py

```diff
@@ -49,12 +48,14 @@ def _get_dep_statuses(self, ti, session, dep_context):
         finished_task_ids = {t.task_id for t in finished_tis}
 
         for parent in upstream:
-            if isinstance(parent, SkipMixin):
+            if parent.inherits_from_skip_mixin is True:
                 if parent.task_id not in finished_task_ids:
                     # This can happen if the parent task has not yet run.
                     continue
 
-                prev_result = ti.xcom_pull(task_ids=parent.task_id, key=XCOM_SKIPMIXIN_KEY, session=session)
+                # TODO: Use XCom.deserialize_value instead (requires some additional adjustments)
+                import json
+                prev_result = json.loads(ti.xcom_pull(task_ids=parent.task_id, key=XCOM_SKIPMIXIN_KEY, session=session) or "{}")
```

Comment:
`xcom_pull` should already deserialize, but there's a bug where we are double encoding things. Please leave a todo comment linking to #45231 here.

cc @amoghrajesh in case this gets merged before you fix that.

Reply:
It seems to work now, so I removed the `json.loads`.
I added the TODO comment though.